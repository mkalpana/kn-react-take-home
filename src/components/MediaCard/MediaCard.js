// @flow
import React from 'react';
import styled from 'styled-components';
import { media } from '../../utils';

type Props = {
  image: string,
  imageAltText: string,
  logo: string,
  logoAltText: string,
  description: string,
};

const Container = styled.div`
  position: relative;
`;


const Image = styled.img`
  border: 1em solid #010101;
  box-sizing: border-box;
  width: 768px;
  
  ${media.tablet`
    width: 100%;
  `}
  ${media.phone`
    width: 100%;
  `}
`;

const Title = styled.div`
  position: absolute;
  bottom: 4px;
  margin-bottom: 1em;
  display: flex;
  align-items: center;
  background: rgba(24, 24, 24, 0.5);
  width: 48em;
  box-sizing: border-box;
  
  ${media.tablet`
    width: 100%;
  `}
  ${media.phoneLarge`
    width: 100%;
  `}
  ${media.phone`
    width: 100%;
  `}
`;

const Logo = styled.img`
  margin: 0 1em 0 1.5em;
  width: 6.25em;
  height: 6.25em;
  
  ${media.tablet`
    width: 5em;
    height: 5em;
  `}
  ${media.phoneLarge`
    width: 2.625em;
    height: 2.625em;
  `}
  ${media.phone`
    margin: 0 0.75em 0 1.5em;
    width: 2em;
    height: 2em;
  `}
`;

const Description = styled.h2`
  font-size: 3em;
  font-weight: 500;
  color: white;
  
  ${media.tablet`
    font-size: 2em;
  `}
  ${media.phoneLarge`
    font-size: 1.25em;
  `}
  ${media.phone`
    font-size: 1em;
  `}
`;

const MediaCard = (props: Props) => {
  const { image, imageAltText, logo, logoAltText, description } = props;
  return (
    <Container>
      <Image src={image} alt={imageAltText} />
      <Title>
        <Logo src={logo} alt={logoAltText}/>
        <Description>{description}</Description>
      </Title>
    </Container>
  );
};

export default MediaCard;
