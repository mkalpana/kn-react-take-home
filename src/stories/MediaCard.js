import React from 'react';
import { storiesOf } from '@storybook/react';
import { MediaCard } from '../components';
import tile from '../img/tile.jpg';
import logo from '../img/logo.png';

storiesOf('MediaCard', module)
  .add('default view', () => (
    <MediaCard
      image={tile}
      imageAltText="Home and Away"
      logo={logo}
      logoAltText="Seven"
      description="Home and Away"
    />
  ));
