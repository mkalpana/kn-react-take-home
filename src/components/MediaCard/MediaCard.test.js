import React from 'react';
import { shallow } from 'enzyme';
import MediaCard from './MediaCard';
import tile from '../../img/tile.jpg';
import logo from '../../img/logo.png';

it('renders without crashing.', () => {
  shallow(<MediaCard />);
});

it('should match snapshot..', () => {
  const wrapper = shallow(<MediaCard
    image={tile}
    imageAltText="Home and Away"
    logo={logo}
    logoAltText="Seven"
    description="Home and Away"
  />);
  expect(wrapper).toMatchSnapshot();
});
