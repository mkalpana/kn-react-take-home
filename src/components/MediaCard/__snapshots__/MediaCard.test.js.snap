// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot.. 1`] = `
<styled.div>
  <styled.img
    alt="Home and Away"
    src="tile.jpg"
  />
  <styled.div>
    <styled.img
      alt="Seven"
      src="logo.png"
    />
    <styled.h2>
      Home and Away
    </styled.h2>
  </styled.div>
</styled.div>
`;
